### PR TITLE
Fix decal

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15323/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15323/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15323/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15323/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15323/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15323/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15323/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15323/merge#BCU1XR#0

File: packages/dev/core/src/Meshes/Builders/decalBuilder.ts [hunk 1]
```diff
@@ -381,13 +381,8 @@ export function CreateDecal(
...
             faceVertices[0] = extractDecalVector3(index, transformMatrix);
-            if (useLocalComputation) {
-                faceVertices[1] = extractDecalVector3(index + 2, transformMatrix);
-                faceVertices[2] = extractDecalVector3(index + 1, transformMatrix);
-            } else {
-                faceVertices[1] = extractDecalVector3(index + 1, transformMatrix);
-                faceVertices[2] = extractDecalVector3(index + 2, transformMatrix);
-            }
+            faceVertices[1] = extractDecalVector3(index + 1, transformMatrix);
+            faceVertices[2] = extractDecalVector3(index + 2, transformMatrix);
```
Comment (Popov72) [inline packages/dev/core/src/Meshes/Builders/decalBuilder.ts:385]: I wonder why the existing code does not work anymore after the PR on the side orientation...

Comment (deltakosh) [reply packages/dev/core/src/Meshes/Builders/decalBuilder.ts:385]: Haha I was wondering WHY we needed that in the first hand. why do we need to change winding order if using local data?

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15323/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15323/merge/testResults/webgpuplaywright/index.html

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Meshes/Builders/decalBuilder.ts (hunk 1 above)
Comment (Popov72) [reply packages/dev/core/src/Meshes/Builders/decalBuilder.ts:385]: No idea, but it worked :)

**Comment by redstair** [comment]
This breaks my implementation of decals on moving objects. Decals seem to get parented to the mesh but are not visible.

BABYLON.MeshBuilder.CreateDecal(`decal_${id.toString()}`, sourceMesh, {
      position: pos,
      normal: normal,
      angle: angle,
      cullBackFaces: false,
      localMode: true,
      size: new BABYLON.Vector3(1, 1, 1),
    })

**Comment by sebavan** [comment]
@redstair can you share a repro ?

**Comment by redstair** [comment]
Wasn't able to reproduce this in playground ... solved by setting backFaceCulling = false
worked without in 7.16
updated to 7.17.2 decals were gone, had to set 
`decal.material.backFaceCulling = false`
-> visible again

It looks like decal facing for localMode was flipped. This may be what the removed two lines were for :)
Thanks for your time!

**Pull request merged**